source/dub/git.d

```diff
@@ -0,0 +1,47 @@
+// functionality to supply packages from git submodules
+module dub.git;
+
+import dub.dependency;
+import dub.internal.vibecompat.core.file;
+import dub.package_;
+import dub.packagemanager;
+
+import std.algorithm;
+import std.ascii : newline;
+import std.exception : enforce;
+import std.range;
+import std.string;
+
+/** Adds the git submodules checked out in the root path as direct packages.
+	Package version is derived from the submodule's tag by `getOrLoadPackage`.
+
+	Params:
+		packageManager = Package manager to track the added packages.
+		rootPath = the root path of the git repository to check for submodules.
+ */
+public void addGitSubmodules(PackageManager packageManager, NativePath rootPath) {
+	import std.process : execute;
+
+	auto rootScmPath = rootPath ~ ".git";
+	const submoduleInfo = execute([
+		"git",
+		"-C", rootPath.toNativeString,
+		"--git-dir=" ~ (rootScmPath.relativeTo(rootPath)).toNativeString,
```

Comment:
Why is `--git-dir` necessary here? It was necessary in conjunction with `--work-tree` before `-C` was introduced.